src/api/apiMethod.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import { GET } from 'constants';
+import { solveURLParams, toQueryString } from './utils';
+
+// Bind to resource class
+const apiMethod = (params = {}) => {
+	const {
+		method = GET,
+		path = '',
+		urlParams = [],
+		validator = null,
+		defaultData = {},
+		retry = false,
+	} = params;
```

Comment:
Why not do this directly in the parameters brackets for the function?

Reply:
```
const apiMethod = ({
		method = GET,
		path = '',
		urlParams = [],
		validator = null,
		defaultData = {},
		retry = false,
	}) => function apiHandler(...args) {
```
wouldn't it be harder to read?

Reply:
For me it's easier, but happy to defer to the majority if @Tosch110 agrees with you.

Reply:
@Tosch110 I don't understand that point about shorthand property assignment?